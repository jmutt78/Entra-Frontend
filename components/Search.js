import React, { useState, useEffect } from 'react';
import capitalize from 'lodash/capitalize';
import { withRouter } from 'next/router';

import IconButton from '@material-ui/core/IconButton';
import InputBase from '@material-ui/core/InputBase';
import Menu from '@material-ui/core/Menu';
import MenuIcon from '@material-ui/icons/Menu';
import MenuItem from '@material-ui/core/MenuItem';
import Paper from '@material-ui/core/Paper';
import SearchIcon from '@material-ui/icons/Search';
import { makeStyles } from '@material-ui/core/styles';

import { usePageContext } from './layout';

const useStyles = makeStyles(theme => ({
  root: {
    padding: '2px 4px',
    display: 'flex',
    alignItems: 'center',
    width: 400
  },
  input: {
    marginLeft: theme.spacing(1),
    flex: 1
  },
  iconButton: {
    padding: 10
  },
  divider: {
    height: 28,
    margin: 4
  }
}));

const CustomizedInputBase = ({ router }) => {
  const classes = useStyles();
  const [fieldState, setFieldState] = useState('');
  const { setSearchTerm, setSearchScope, searchScope } = usePageContext();
  const [anchorEl, setAnchorEl] = useState(null);

  const handleMenuClick = ({ currentTarget }) => {
    setAnchorEl(currentTarget);
  };

  const handleMenuClose = () => {
    setAnchorEl(null);
  };

  const handMenuSelect = scope => {
    setSearchScope(scope);
    handleMenuClose();
  };

  const handleKeyPress = event => {
    if (event.key === 'Enter') {
      setSearchTerm(fieldState);
    }
  };

  return (
    <>
      <Paper className={classes.root}>
        <IconButton
          className={classes.iconButton}
          aria-label="menu"
          onClick={handleMenuClick}
        >
          <MenuIcon />
        </IconButton>
        <InputBase
          className={classes.input}
          placeholder={`Search ${capitalize(searchScope)}`}
          inputProps={{ 'aria-label': 'search question' }}
          onChange={({ target: { value } }) => setFieldState(value)}
          onKeyPress={handleKeyPress}
        />
        <IconButton
          className={classes.iconButton}
          aria-label="search"
          onClick={() => setSearchTerm(fieldState)}
        >
          <SearchIcon />
        </IconButton>
      </Paper>
      <Menu
        anchorEl={anchorEl}
        open={Boolean(anchorEl)}
        onClose={handleMenuClose}
      >
        <MenuItem onClick={() => handMenuSelect('all')}>Search All</MenuItem>
        <MenuItem onClick={() => handMenuSelect('titles')}>
          Search Titles
        </MenuItem>
        <MenuItem onClick={() => handMenuSelect('authors')}>
          Search Authors
        </MenuItem>
        <MenuItem onClick={() => handMenuSelect('tags')}>Search Tags</MenuItem>
      </Menu>
    </>
  );
};

export default withRouter(CustomizedInputBase);
