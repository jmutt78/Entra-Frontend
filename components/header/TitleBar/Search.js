import React from 'react';

import DirectionsIcon from '@material-ui/icons/Directions';
import Divider from '@material-ui/core/Divider';
import IconButton from '@material-ui/core/IconButton';
import InputBase from '@material-ui/core/InputBase';
import MenuIcon from '@material-ui/icons/Menu';
import Paper from '@material-ui/core/Paper';
import SearchIcon from '@material-ui/icons/Search';
import SortIcon from '@material-ui/icons/Sort';
import { makeStyles } from '@material-ui/core/styles';

const useStyles = makeStyles(theme => ({
  input: {
    marginLeft: theme.spacing(1),
    flex: 1
  },
  iconButton: {
    padding: 10
  },
  divider: {
    height: 28,
    margin: 4
  }
}));

const Search = () => {
  const classes = useStyles();

  return (
    <Paper className="searchField">
      <IconButton className={classes.iconButton} aria-label="menu">
        <SortIcon />
      </IconButton>
      <InputBase
        className={classes.input}
        placeholder="Search Questions"
        inputProps={{ 'aria-label': 'search google maps' }}
      />
      <IconButton className={classes.iconButton} aria-label="search">
        <SearchIcon />
      </IconButton>
      <Divider className={classes.divider} orientation="vertical" />
    </Paper>
  );
};

export default Search;
