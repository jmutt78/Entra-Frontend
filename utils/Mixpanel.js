import mixpanel from 'mixpanel-browser';
mixpanel.init('bfc78222da8bc97ec71a9e3d053adab6');

let actions = {
  identify: id => {
    mixpanel.identify(id);
  },
  alias: id => {
    mixpanel.alias(id);
  },
  track: (name, props) => {
    mixpanel.track(name, props);
  },
  people: {
    set: props => {
      mixpanel.people.set(props);
    }
  }
};

export let Mixpanel = actions;
