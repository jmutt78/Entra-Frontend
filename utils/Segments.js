import React from 'react';
import Head from 'next/head';

import SEGMENT_PIXEL_1 from './segment-1.js';

export default ({ name }) => {
  return <Head>{name === 'SEGMENT_PIXEL_1' && <SEGMENT_PIXEL_1 />}</Head>;
};
