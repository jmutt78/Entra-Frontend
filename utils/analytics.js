import ReactGA from 'react-ga';
const google = 'UA-145851763-1';

export const initGA = () => {
  ReactGA.initialize(google);
};
export const logPageView = () => {
  ReactGA.set({ page: window.location.pathname });
  ReactGA.pageview(window.location.pathname);
};
export const logEvent = (category = '', action = '') => {
  if (category && action) {
    ReactGA.event({ category, action });
  }
};
export const logException = (description = '', fatal = false) => {
  if (description) {
    ReactGA.exception({ description, fatal });
  }
};
