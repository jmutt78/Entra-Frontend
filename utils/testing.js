const fakeQuestion = {
  __typename: 'Question',
  answers: [],
  approval: true,
  askedBy: [],
  bookMark: [],
  createdAt: '2019-11-08T08:20:54.504Z',
  description: '',
  downVotes: 3,
  id: '5dc525660274390007608a12',
  searchTermFoundIn: 'Question',
  tags: [],
  title: 'test1',
  upVotes: 6,
  views: 9
};

const fakeAnswer = {
  __typename: 'Answer',
  body: 'answer test 1',
  id: '5db293ee857aba00085589c9'
};

const fakeAskedby = {
  __typename: 'User',
  display: 'userTest1',
  id: '5db289a3857aba00085589b6',
  name: 'ut1'
};

const fakeBookmark = {
  __typename: 'BookMark',
  id: '5ddbdd360274390007608a2f'
};

const fakseTags = {
  __typename: 'Tag',
  id: '5db29080857aba00085589bb',
  name: 'testTag1'
};

export { fakeQuestion, fakseTags, fakeBookmark, fakeAskedby, fakeAnswer };
