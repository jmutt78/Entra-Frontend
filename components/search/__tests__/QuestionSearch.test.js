import React from 'react';
import { shallow, mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import Search, { SEARCH_QUESTIONS_QUERY } from '../QuestionSearch';
import wait from 'waait';
import { MockedProvider } from 'react-apollo/test-utils';
import { fakeQuestion, fakeAskedby } from '../../../utils/testing';

describe('<Search />', () => {
  it('renders', () => {
    shallow(<Search />);
  });

  it('matches the snapshot', () => {
    const wrapper = shallow(<Search />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('renders with dropdown', async () => {
    const mocks = [
      {
        request: {
          query: SEARCH_QUESTIONS_QUERY,
          variables: {
            searchTerm: 'test'
          }
        },
        result: {
          data: {
            searchQuestions: [
              {
                ...fakeQuestion
              }
            ]
          }
        }
      }
    ];
    const wrapper = mount(
      <MockedProvider mocks={mocks}>
        <Search />
      </MockedProvider>
    );
    wrapper.find('input').simulate('change', { target: { value: 'test' } });
    await wait(400);
    wrapper.update();
    expect(toJSON(wrapper.find('QuestionSearch__DropDown'))).toMatchSnapshot();
  });

  it('renders with proper title in dropdown item', async () => {
    const mocks = [
      {
        request: {
          query: SEARCH_QUESTIONS_QUERY,
          variables: {
            searchTerm: 'test'
          }
        },
        result: {
          data: {
            searchQuestions: [
              {
                ...fakeQuestion,
                title: `${fakeAskedby.display} asked`,
                askedBy: [{ ...fakeAskedby }]
              }
            ]
          }
        }
      }
    ];
    const wrapper = mount(
      <MockedProvider mocks={mocks}>
        <Search />
      </MockedProvider>
    );
    wrapper.find('input').simulate('change', { target: { value: 'test' } });
    await wait(400);
    wrapper.update();
    expect(wrapper.find('QuestionSearch__DropDown').text()).toContain(
      `${fakeAskedby.display} asked`
    );
  });

  it('renders with More results dropdown item', async () => {
    const mocks = [
      {
        request: {
          query: SEARCH_QUESTIONS_QUERY,
          variables: {
            searchTerm: 'test'
          }
        },
        result: {
          data: {
            searchQuestions: [
              {
                ...fakeQuestion,
                id: '1'
              },
              {
                ...fakeQuestion,
                id: '2'
              },
              {
                ...fakeQuestion,
                id: '3'
              },
              {
                ...fakeQuestion,
                id: '4'
              },
              {
                ...fakeQuestion,
                id: '5'
              },
              {
                ...fakeQuestion,
                id: '6'
              }
            ]
          }
        }
      }
    ];
    const wrapper = mount(
      <MockedProvider mocks={mocks}>
        <Search />
      </MockedProvider>
    );
    wrapper.find('input').simulate('change', { target: { value: 'test' } });
    await wait(400);
    wrapper.update();
    expect(wrapper.find('QuestionSearch__DropDown').text()).toContain(
      'More results for test'
    );
  });
});
