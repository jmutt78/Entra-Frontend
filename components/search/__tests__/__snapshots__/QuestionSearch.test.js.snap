// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Search /> matches the snapshot 1`] = `
<QuestionSearch__SearchContainer>
  <Downshift
    defaultHighlightedIndex={null}
    defaultInputValue=""
    defaultIsOpen={false}
    defaultSelectedItem={null}
    environment={[Window]}
    getA11yStatusMessage={[Function]}
    itemToString={[Function]}
    onChange={[Function]}
    onInputValueChange={[Function]}
    onOuterClick={[Function]}
    onSelect={[Function]}
    onStateChange={[Function]}
    onUserAction={[Function]}
    scrollIntoView={[Function]}
    selectedItemChanged={[Function]}
    stateReducer={[Function]}
    suppressRefError={false}
  >
    <Component />
  </Downshift>
</QuestionSearch__SearchContainer>
`;

exports[`<Search /> renders with dropdown 1`] = `
<QuestionSearch__DropDown>
  <div
    className="QuestionSearch__DropDown-sc-1kwr300-1 kgdCcu"
  >
    <QuestionSearch__DropDownItem
      aria-selected={false}
      highlighted={false}
      id="downshift-0-item-0"
      key="5dc525660274390007608a12:test1"
      onClick={[Function]}
      onMouseDown={[Function]}
      onMouseMove={[Function]}
      role="option"
    >
      <div
        aria-selected={false}
        className="QuestionSearch__DropDownItem-sc-1kwr300-2 jTRfYK"
        id="downshift-0-item-0"
        onClick={[Function]}
        onMouseDown={[Function]}
        onMouseMove={[Function]}
        role="option"
      >
        Question: test1
      </div>
    </QuestionSearch__DropDownItem>
  </div>
</QuestionSearch__DropDown>
`;
